app.js

```diff
@@ -0,0 +1,157 @@
+//jshint esversion: 6
+const express = require("express");
+const bodyParser = require("body-parser");
+const mongoose = require("mongoose");
+const redisClient = require("redis").createClient;
+const redis = redisClient(6379, 'redis-server');
+
+const app = express();
+
+app.set('view engine', 'ejs');
+app.use(bodyParser.urlencoded({
+  extended: true
+}));
+app.use(express.static("public"));
+//let studentMap = new Map();
+
+mongoose.connect("mongodb://mongo:27017/mongorediscomposeDB", {
```

Comment:
The connection params can become part of a config file, try to divide the responsibilities and deal with only parameters everywhere so that code becomes more open in future. Try to think of a code in classes and methods and then there should be driver programs or final applications which uses these methods and classes as libraries.

So you can may be then use the same method for making a CLI application or a web server etc.